tests/spark/autologging/ml/test_pyspark_ml_autologging.py

```diff
@@ -323,6 +323,55 @@ def test_should_log_model(dataset_binomial, dataset_multinomial, dataset_text):
         assert not _should_log_model(nested_pipeline_model)
 
 
+def test_log_stage_type_params(spark_session):
+    from pyspark.ml.base import Estimator, Transformer, Model
+    from pyspark.ml.param import Param, Params
+    from pyspark.ml.feature import Binarizer, OneHotEncoder
+
+    class TestingEstimator(Estimator):
+
+        transformer = Param(Params._dummy(), "transformer", "a transformer param")
+        model = Param(Params._dummy(), "model", "a model param")
+
+        def setTransformer(self, transformer: Transformer):
+            return self._set(transformer=transformer)
+
+        def setModel(self, model: Model):
+            return self._set(model=model)
+
+        def _fit(self, dataset):
+            return TestingModel()
+
+    class TestingModel(Model):
+        def _transform(self, dataset):
+            return dataset
+
+    binarizer = Binarizer(threshold=1.0, inputCol="values", outputCol="features")
+    df = spark_session.createDataFrame([(0.0,), (1.0,), (2.0,)], ["input"])
+    ohe = OneHotEncoder().setInputCols(["input"]).setOutputCols(["output"])
+    ohemodel = ohe.fit(df)
+
+    estimator = TestingEstimator().setTransformer(binarizer).setModel(ohemodel)
+    param_map = get_params_to_log(estimator)
+    assert param_map["transformer"] == "Binarizer"
+    assert param_map["model"] == "OneHotEncoderModel"
+
+    mlflow.pyspark.ml.autolog()
+    with mlflow.start_run() as run:
+        estimator.fit(df)
+        metadata = _gen_estimator_metadata(estimator)
+        estimator_info = load_json_artifact("estimator_info.json")
+        assert metadata.hierarchy == estimator_info["hierarchy"]
+        assert isinstance(estimator_info["hierarchy"]["params"], dict)
+        assert estimator_info["hierarchy"]["params"]["transformer"]["name"] == "Binarizer"
+        assert estimator_info["hierarchy"]["params"]["model"]["name"] == "OneHotEncoderModel"
+    run_id = run.info.run_id
+    run_data = get_run_data(run_id)
+    assert run_data.params == truncate_param_dict(
+        stringify_dict_values(get_params_to_log(estimator))
+    )
```

Comment:
I suggest also add "Evaluator" type param case into the test.

Reply:
done.